src/Augurk/Managers/FeatureManager.cs

```diff
@@ -307,24 +307,40 @@ public async Task<DbFeature> InsertOrUpdateFeatureAsync(Feature feature, string
 
             using (var session = _storeProvider.Store.OpenAsyncSession())
             {
-
                 dbFeature = await session.LoadAsync<DbFeature>(DbFeatureExtensions.GetIdentifier(productName, groupName, feature.Title, hash));
 
-                if(dbFeature != null){
+                if (dbFeature != null)
+                {
                     // Add the new version to the list
                     var versions = new List<string>(dbFeature.Versions);
                     versions.Add(version);
                     // Prevent duplicates
                     dbFeature.Versions = versions.Distinct().ToArray();
                 }
-                else{
+                else
+                {
                     // Create a new feature
                     var processor = new FeatureProcessor();
                     string parentTitle = processor.DetermineParent(feature);
-                    dbFeature = new DbFeature(feature, productName, groupName, parentTitle, version);
+                    dbFeature = new DbFeature(feature, productName, groupName, parentTitle, version) { Hash = hash };
                     await session.StoreAsync(dbFeature, dbFeature.GetIdentifier());
                 }
 
+                var wronglyStampedFeatures = session.Query<DbFeature, Features_ByTitleProductAndGroup>()
```

Comment:
Some comment for explaination might not be amiss

Reply:
Fixed